TYPO3.Neos/Classes/TYPO3/Neos/TypoScript/DimensionsMenuImplementation.php

```diff
@@ -122,6 +138,20 @@ protected function buildItems()
             ];
         }
 
+        // sort/limit according to configured "presets" if needed
+        if ($pinnedDimensionName !== null && is_array($pinnedDimensionValues)) {
+            $sortedMenuItems = [];
+            foreach ($pinnedDimensionValues as $pinnedDimensionValue) {
+                foreach ($menuItems as $menuItemKey => $menuItem) {
+                    if ($menuItem['targetDimensions'][$pinnedDimensionName]['value'] === $pinnedDimensionValue) {
+                        $sortedMenuItems[$menuItemKey] = $menuItem;
+                    }
+                }
+            }
```

Comment:
Can't this be done with some kind of usort that compares the index value of the `$pinnedDimensionValue`?
I think that would be more explicit about what's going on and also a bit less complex. I can give it a try.